File: app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
ArrayList<RecyclerViewItem>recyclerViewItems;
private RecyclerView recyclerView;
private RecyclerView.Adapter adapter;
private RecyclerView.LayoutManager layoutManager;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerViewItems=new ArrayList<>();
        recyclerViewItems.add(new RecyclerViewItem(R.drawable.ic_android_black_24dp,"Poxos","Poxosya"));
        recyclerViewItems.add(new RecyclerViewItem(R.drawable.ic_android_black_24dp,"Poxos","Poxosya"));
        recyclerViewItems.add(new RecyclerViewItem(R.drawable.ic_android_black_24dp,"Poxos","Poxosya"));
        recyclerView=findViewById(R.id.recyclerVi);
        recyclerView.setHasFixedSize(true);
        adapter=new RecyclerViewAdapter(recyclerViewItems);
        layoutManager=new LinearLayoutManager(this) ;
recyclerView.setAdapter(adapter);
recyclerView.setLayoutManager(layoutManager);
    }
}

File: myrecyclerviewkotlin/src/main/java/com/example/myrecyclerviewkotlin/MainActivity.kt
package com.example.myrecyclerviewkotlin

import android.os.Bundle
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
lateinit var adapter: KotRecyclerViewAdapter
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val list= listOf(
                KotReciclerItem(R.drawable.ic_android_black_24dp,"poxos","Poxosyan"),
                KotReciclerItem(R.drawable.ic_android_black_24dp,"poxos","Poxosyan"),
                KotReciclerItem(R.drawable.ic_android_black_24dp,"poxos","Poxosyan")
        )
        adapter=KotRecyclerViewAdapter(list as MutableList<KotReciclerItem>)
val layoutManager=LinearLayoutManager(this)
       recyclerView.setHasFixedSize(true)
        recyclerView.layoutManager=layoutManager
        recyclerView.adapter=adapter
    }
}
